README
This RTAMonitor it's a simple text based server that prints to output the parameters and logs received from clients.

To build/install just run:

  $ make
  $ make install

To run you need to type:

  $ ./RTAMonitorQt --Ice.Config=config.monitor


ChangeLog
2014-12-12 Andrea Zoli <zoli@iasfbo.inaf.it>
TAG v1.0.2
* Use the right IceUtil naming for Darwin.
* Fix Make.rules.Linux to handle clang++.
* Use conditional assignment for CPPFLAGS in Makefile.

2014-07-25 Andrea Zoli <zoli@iasfbo.inaf.it>
TAG v1.0.1
* Add make install.
* Looking for slice files from $CTARTA.
* README updated.

2014-04-24 Andrea Zoli<zoli@iasfbo.inaf.it>, Andrea Bulgarelli <bulgarelli@iasfbo.inaf.it>
TAG v1.0.0
* -lIce to -lZerocIce fixed
* changes on Gui.ui
* RTAMonitorQt: added camera/s and event/s
* Added *.o to the .gitignore.
* Changed sligtly the Gui for RTAMonitorQt.
* Gui changes
* Range 0-3000 usecs for RTAMonitorQt slider.
* Added Qt4 Designer .ui file (related to RTAMonitorQt).
* RTAMonitorQt extend Ui_Monitor and implements RTAMonitor behaviour.
* Add RTAMonitorQt.py stub.
* Add registerApp to RTAMonitorI.
* RTAMonitor.py: Added RTAReceiver rate.
* RTAMonitor.py sends new delay to RTAEBSim.
* RTAMonitor.py Visualize EBSim rate.
* Add Command interface to RTAMonitor.py.
* Add RTAMonitor.py.
* Parameter type 0 stands for rate (MB/s).
* Update README.
* Change config.monitor port to 20000.
* Fix _parameters and _logs names.
* Initial commit.


RTAMonitor.py
#!/usr/bin/env python

############################################################################
#    begin                : Mar 25 2014
#    copyright            : (C) 2014 Andrea Zoli, Valentina Fioretti
#    email                : zoli@iasfbo.inaf.it, fioretti@iasfbo.inaf.it
############################################################################

############################################################################
#                                                                          #
#   This program is free software for non commercial purpose               #
#   and for public research institutes; you can redistribute it and/or     #
#   modify it under the terms of the GNU General Public License.           #
#   For commercial purpose see appropriate license terms                   #
#                                                                          #
############################################################################

import sys, traceback, time, Ice

confdir=os.environ["CTARTA"]+"/share/monitor/"

Ice.loadSlice(confdir+"RTAMonitor.ice")
Ice.loadSlice(confdir+"RTACommand.ice")

Ice.updateModules()
import CTA

from traits.api \
    import HasTraits, Range, Float, String, on_trait_change, Property
from traitsui.api import View, Item

class MonitorI(HasTraits, CTA.RTAMonitor):

    simDelay = Range(0.0, 2000.0)
    simRate = String("0 MB/s")
    receiverRate = String("0 MB/s")

    traits_view = View(Item(name='simDelay', style='simple'),
                       Item(name='simRate', style='readonly'),
                       Item(name='receiverRate', style='readonly'),
                       resizable=True,
                       title='RTA Sleep',
                       width=500, height=200)

    def __init__(self, ic):
        HasTraits.__init__(self)
        CTA.RTAMonitor.__init__(self)
        self.ic = ic
        self.command = 0

    def _simDelay_changed(self):
        if self.command != 0:
            self.command.setSimDelay(self.simDelay)

    def sendParameter(self, param, current=None):
#        print "apid:", param.apid, "type:", param.type, "time:", param.timestamp, "rate:", param.value, "MB/s"
        if param.type == 0:
            if param.apid == 200:
                self.simRate = str(param.value)+" MB/s"
            if param.apid == 201:
                self.receiverRate = str(param.value)+" MB/s"

    def sendLog(self, msg, current=None):
        print(msg.apid, msg.timestamp, msg.value)

    def registerApp(self, apid, current=None):
        if apid == 200:
            print "RTAEBSim registered."
            self.command = CTA.RTACommandPrx.checkedCast(self.ic.propertyToProxy('RTAEBSimCommand.Proxy')).ice_oneway()

class Monitor(Ice.Application):
    def run(self, args):
        if len(args) > 1:
            print(self.appName() + ": too many arguments")
            return 1

		# Activate RTAMonitor adapter
        ic = self.communicator()
        adapter = ic.createObjectAdapter("RTAMonitor")
        servant = MonitorI(ic)
        adapter.add(servant, ic.stringToIdentity("monitor"))
        adapter.activate()

        servant.configure_traits()
        return 0

if __name__ == "__main__":
    monitor = Monitor()
    sys.exit(monitor.main(sys.argv))


RTAMonitorI.h
/***************************************************************************
    begin                : Mar 13 2014
    copyright            : (C) 2014 Andrea Zoli
    email                : zoli@iasfbo.inaf.it
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software for non commercial purpose              *
 *   and for public research institutes; you can redistribute it and/or    *
 *   modify it under the terms of the GNU General Public License.          *
 *   For commercial purpose see appropriate license terms                  *
 *                                                                         *
 ***************************************************************************/

#ifndef _RTAMONITOR_I_H
#define _RTAMONITOR_I_H

#include "RTAMonitor.h"

class RTAMonitorI : public CTA::RTAMonitor
{
public:

	RTAMonitorI(std::map<int, std::vector<CTA::Parameter> >& parameters, 
                std::map<int, std::vector<CTA::LogMessage> >& logs)
		: _parameters(parameters), _logs(logs)
	{
	}
	virtual void sendParameter(const CTA::Parameter& param, const Ice::Current& cur);
	virtual void sendLog(const CTA::LogMessage& msg, const Ice::Current& cur);

	virtual void registerApp(int apid, const Ice::Current& cur);

private:

	std::map<int, std::vector<CTA::Parameter> >& _parameters;
	std::map<int, std::vector<CTA::LogMessage> >& _logs;

};

#endif


RTAMonitorI.cpp
/***************************************************************************
    begin                : Mar 13 2014
    copyright            : (C) 2014 Andrea Zoli
    email                : zoli@iasfbo.inaf.it
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software for non commercial purpose              *
 *   and for public research institutes; you can redistribute it and/or    *
 *   modify it under the terms of the GNU General Public License.          *
 *   For commercial purpose see appropriate license terms                  *
 *                                                                         *
 ***************************************************************************/

#include "RTAMonitorI.h"
#include <iostream>
#include <iomanip>

using namespace CTA;

void RTAMonitorI::sendParameter(const Parameter& param, const Ice::Current& cur)
{
	_parameters[param.apid].push_back(param);
	if(param.type == 0) // rate
	{
		std::cout << "apid: " << param.apid << " time: " << std::setprecision(6) << param.timestamp << " rate: " << param.value << " MB/s" << std::endl;
	}
	if(param.type == 1) // camera/s
	{
		std::cout << "apid: " << param.apid << " time: " << std::setprecision(6) << param.timestamp << " rate: " << param.value << " camera/s" << std::endl;
	}
	if(param.type == 2) // event/s
	{
		std::cout << "apid: " << param.apid << " time: " << std::setprecision(6) << param.timestamp << " rate: " << param.value << " event/s" << std::endl;
	}

}

void RTAMonitorI::sendLog(const LogMessage& msg, const Ice::Current& cur)
{
	_logs[msg.apid].push_back(msg);
	std::cout << msg.apid << " " << msg.timestamp << " " << msg.value << std::endl;
}

void RTAMonitorI::registerApp(int apid, const Ice::Current& cur)
{
	std::cout << "Registered APID " << apid << std::endl;
}


RTAMonitorQt.py
#!/usr/bin/env python

############################################################################
#    begin                : Mar 25 2014
#    copyright            : (C) 2014 Andrea Zoli
#    email                : zoli@iasfbo.inaf.it
############################################################################

############################################################################
#                                                                          #
#   This program is free software for non commercial purpose               #
#   and for public research institutes; you can redistribute it and/or     #
#   modify it under the terms of the GNU General Public License.           #
#   For commercial purpose see appropriate license terms                   #
#                                                                          #
############################################################################

from PyQt4.QtGui import QApplication, QDialog
from PyQt4 import QtCore
from Gui import Ui_Monitor

import os, sys, traceback, time, Ice

confdir=os.environ["CTARTA"]+"/share/monitor/"

Ice.loadSlice(confdir+"RTAMonitor.ice")
Ice.loadSlice(confdir+"RTACommand.ice")

Ice.updateModules()
import CTA

class MonitorI(CTA.RTAMonitor, Ui_Monitor):
    simRate = 0.0
    receiverRate = 0.0

    def __init__(self, ic):
        CTA.RTAMonitor.__init__(self)
        self.ic = ic
        self.command = 0

    def sendParameter(self, param, current=None):
        print "apid:", param.apid, "type:", param.type, "time:", param.timestamp, "rate:", param.value, "MB/s"
        if param.type == 0:
            if param.apid == 200:
                self.label_6.setText(str(int(param.value)))
            if param.apid == 201:
                self.label_9.setText(str(int(param.value)))
	
        elif param.type == 1:
            if param.apid == 200:
                self.label_14.setText(str(int(param.value)))
        elif param.type == 2:
	        if param.apid == 200:
		        self.label_15.setText(str(int(param.value)))
		
    def sendLog(self, msg, current=None):
        print(msg.apid, msg.timestamp, msg.value)

    def registerApp(self, apid, current=None):
        if apid == 200:
            print "RTAEBSim registered."
            self.command = CTA.RTACommandPrx.checkedCast(self.ic.propertyToProxy('RTAEBSimCommand.Proxy')).ice_oneway()

    def setupUi(self, Monitor):
        Ui_Monitor.setupUi(self, Monitor)
        QtCore.QObject.connect(self.horizontalSlider, QtCore.SIGNAL("valueChanged(int)"), self.updateLabel)
        QtCore.QObject.connect(self.lineEdit, QtCore.SIGNAL("textEdited(const QString&)"), self.updateSlider)

        self.horizontalSlider.setMinimum(0)
        self.horizontalSlider.setMaximum(3000)

    def updateSlider(self, val):
        self.horizontalSlider.setValue(int(val))
        if self.command != 0:
            self.command.setSimDelay(int(val))

    def updateLabel(self, val):
        self.lineEdit.setText(str(val))
        if self.command != 0:
            self.command.setSimDelay(val)

class Monitor(Ice.Application):
    def run(self, args):
        if len(args) > 1:
            print(self.appName() + ": too many arguments")
            return 1

		# Activate RTAMonitor adapter
        ic = self.communicator()
        adapter = ic.createObjectAdapter("RTAMonitor")
        servant = MonitorI(ic)
        adapter.add(servant, ic.stringToIdentity("monitor"))
        adapter.activate()

        app = QApplication(sys.argv)
        window = QDialog()
        servant.setupUi(window)

        window.show()
        sys.exit(app.exec_())

if __name__ == "__main__":
    monitor = Monitor()
    sys.exit(monitor.main(sys.argv))


RTAMonitorServer.cpp
/***************************************************************************
    begin                : Mar 13 2014
    copyright            : (C) 2014 Andrea Zoli
    email                : zoli@iasfbo.inaf.it
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software for non commercial purpose              *
 *   and for public research institutes; you can redistribute it and/or    *
 *   modify it under the terms of the GNU General Public License.          *
 *   For commercial purpose see appropriate license terms                  *
 *                                                                         *
 ***************************************************************************/

#include <Ice/Ice.h>
#include <IceUtil/IceUtil.h>
#include "RTAMonitorI.h"

using namespace std;
using namespace CTA;

class RTAMonitorServer : public Ice::Application
{
public:

    virtual int run(int, char*[]);

private:

};

int main(int argc, char* argv[])
{
    RTAMonitorServer app;
    return app.main(argc, argv);
}

int RTAMonitorServer::run(int argc, char* argv[])
{
	// parameters indexed per-apid
	std::map<int, std::vector<CTA::Parameter> > parameters;

	// log indexed per-apid
	std::map<int, std::vector<CTA::LogMessage> > logs;

	// Create an adapter for RTAMonitor
    Ice::ObjectAdapterPtr adapter = communicator()->createObjectAdapter("RTAMonitor");
    RTAMonitorPtr servant = new RTAMonitorI(parameters, logs);
    adapter->add(servant, communicator()->stringToIdentity("monitor"));
    adapter->activate();

    communicator()->waitForShutdown();

    return EXIT_SUCCESS;
}


Gui.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'Gui.ui'
#
# Created: Fri Apr  4 12:56:08 2014
#      by: PyQt4 UI code generator 4.10.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui_Monitor(object):
    def setupUi(self, Monitor):
        Monitor.setObjectName(_fromUtf8("Monitor"))
        Monitor.resize(411, 319)
        Monitor.setMaximumSize(QtCore.QSize(800, 400))
        Monitor.setProperty("delay", 0.0)
        self.layoutWidget = QtGui.QWidget(Monitor)
        self.layoutWidget.setGeometry(QtCore.QRect(10, 0, 391, 61))
        self.layoutWidget.setObjectName(_fromUtf8("layoutWidget"))
        self.horizontalLayout = QtGui.QHBoxLayout(self.layoutWidget)
        self.horizontalLayout.setSpacing(30)
        self.horizontalLayout.setSizeConstraint(QtGui.QLayout.SetDefaultConstraint)
        self.horizontalLayout.setContentsMargins(10, -1, 10, -1)
        self.horizontalLayout.setObjectName(_fromUtf8("horizontalLayout"))
        self.label_5 = QtGui.QLabel(self.layoutWidget)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label_5.sizePolicy().hasHeightForWidth())
        self.label_5.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Helvetica"))
        font.setPointSize(24)
        font.setBold(False)
        font.setWeight(50)
        self.label_5.setFont(font)
        self.label_5.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_5.setMargin(10)
        self.label_5.setObjectName(_fromUtf8("label_5"))
        self.horizontalLayout.addWidget(self.label_5)
        self.horizontalSlider = QtGui.QSlider(self.layoutWidget)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.horizontalSlider.sizePolicy().hasHeightForWidth())
        self.horizontalSlider.setSizePolicy(sizePolicy)
        self.horizontalSlider.setAutoFillBackground(False)
        self.horizontalSlider.setMaximum(3000)
        self.horizontalSlider.setPageStep(100)
        self.horizontalSlider.setOrientation(QtCore.Qt.Horizontal)
        self.horizontalSlider.setTickPosition(QtGui.QSlider.NoTicks)
        self.horizontalSlider.setTickInterval(1)
        self.horizontalSlider.setObjectName(_fromUtf8("horizontalSlider"))
        self.horizontalLayout.addWidget(self.horizontalSlider)
        self.lineEdit = QtGui.QLineEdit(self.layoutWidget)
        self.lineEdit.setEnabled(True)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Maximum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.lineEdit.sizePolicy().hasHeightForWidth())
        self.lineEdit.setSizePolicy(sizePolicy)
        self.lineEdit.setMaximumSize(QtCore.QSize(40, 30))
        self.lineEdit.setReadOnly(False)
        self.lineEdit.setObjectName(_fromUtf8("lineEdit"))
        self.horizontalLayout.addWidget(self.lineEdit)
        self.gridLayoutWidget = QtGui.QWidget(Monitor)
        self.gridLayoutWidget.setGeometry(QtCore.QRect(10, 60, 391, 264))
        self.gridLayoutWidget.setObjectName(_fromUtf8("gridLayoutWidget"))
        self.gridLayout = QtGui.QGridLayout(self.gridLayoutWidget)
        self.gridLayout.setSizeConstraint(QtGui.QLayout.SetDefaultConstraint)
        self.gridLayout.setMargin(0)
        self.gridLayout.setObjectName(_fromUtf8("gridLayout"))
        self.label = QtGui.QLabel(self.gridLayoutWidget)
        self.label.setText(_fromUtf8(""))
        self.label.setObjectName(_fromUtf8("label"))
        self.gridLayout.addWidget(self.label, 2, 0, 1, 1)
        self.label_14 = QtGui.QLabel(self.gridLayoutWidget)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label_14.sizePolicy().hasHeightForWidth())
        self.label_14.setSizePolicy(sizePolicy)
        self.label_14.setMaximumSize(QtCore.QSize(150, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(33)
        self.label_14.setFont(font)
        self.label_14.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_14.setMargin(10)
        self.label_14.setObjectName(_fromUtf8("label_14"))
        self.gridLayout.addWidget(self.label_14, 2, 1, 1, 1)
        self.label_10 = QtGui.QLabel(self.gridLayoutWidget)
        self.label_10.setMaximumSize(QtCore.QSize(100, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(26)
        self.label_10.setFont(font)
        self.label_10.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_10.setMargin(10)
        self.label_10.setObjectName(_fromUtf8("label_10"))
        self.gridLayout.addWidget(self.label_10, 1, 2, 1, 1)
        self.label_2 = QtGui.QLabel(self.gridLayoutWidget)
        self.label_2.setText(_fromUtf8(""))
        self.label_2.setObjectName(_fromUtf8("label_2"))
        self.gridLayout.addWidget(self.label_2, 3, 0, 1, 1)
        self.label_9 = QtGui.QLabel(self.gridLayoutWidget)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label_9.sizePolicy().hasHeightForWidth())
        self.label_9.setSizePolicy(sizePolicy)
        self.label_9.setMaximumSize(QtCore.QSize(150, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(43)
        self.label_9.setFont(font)
        self.label_9.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_9.setMargin(10)
        self.label_9.setObjectName(_fromUtf8("label_9"))
        self.gridLayout.addWidget(self.label_9, 1, 1, 1, 1)
        self.label_8 = QtGui.QLabel(self.gridLayoutWidget)
        self.label_8.setMinimumSize(QtCore.QSize(70, 0))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(27)
        self.label_8.setFont(font)
        self.label_8.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_8.setMargin(10)
        self.label_8.setObjectName(_fromUtf8("label_8"))
        self.gridLayout.addWidget(self.label_8, 1, 0, 1, 1)
        self.label_4 = QtGui.QLabel(self.gridLayoutWidget)
        self.label_4.setMinimumSize(QtCore.QSize(70, 0))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(27)
        self.label_4.setFont(font)
        self.label_4.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_4.setMargin(10)
        self.label_4.setObjectName(_fromUtf8("label_4"))
        self.gridLayout.addWidget(self.label_4, 0, 0, 1, 1)
        self.label_6 = QtGui.QLabel(self.gridLayoutWidget)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label_6.sizePolicy().hasHeightForWidth())
        self.label_6.setSizePolicy(sizePolicy)
        self.label_6.setMaximumSize(QtCore.QSize(150, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(43)
        self.label_6.setFont(font)
        self.label_6.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_6.setMargin(10)
        self.label_6.setObjectName(_fromUtf8("label_6"))
        self.gridLayout.addWidget(self.label_6, 0, 1, 1, 1)
        self.label_7 = QtGui.QLabel(self.gridLayoutWidget)
        self.label_7.setMaximumSize(QtCore.QSize(100, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(26)
        self.label_7.setFont(font)
        self.label_7.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_7.setMargin(10)
        self.label_7.setObjectName(_fromUtf8("label_7"))
        self.gridLayout.addWidget(self.label_7, 0, 2, 1, 1)
        self.label_11 = QtGui.QLabel(self.gridLayoutWidget)
        self.label_11.setMaximumSize(QtCore.QSize(100, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(16)
        self.label_11.setFont(font)
        self.label_11.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_11.setMargin(10)
        self.label_11.setObjectName(_fromUtf8("label_11"))
        self.gridLayout.addWidget(self.label_11, 2, 2, 1, 1)
        self.label_15 = QtGui.QLabel(self.gridLayoutWidget)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label_15.sizePolicy().hasHeightForWidth())
        self.label_15.setSizePolicy(sizePolicy)
        self.label_15.setMaximumSize(QtCore.QSize(150, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(33)
        self.label_15.setFont(font)
        self.label_15.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_15.setMargin(10)
        self.label_15.setObjectName(_fromUtf8("label_15"))
        self.gridLayout.addWidget(self.label_15, 3, 1, 1, 1)
        self.label_12 = QtGui.QLabel(self.gridLayoutWidget)
        self.label_12.setMaximumSize(QtCore.QSize(100, 16777215))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("DejaVu Sans"))
        font.setPointSize(16)
        self.label_12.setFont(font)
        self.label_12.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_12.setMargin(10)
        self.label_12.setObjectName(_fromUtf8("label_12"))
        self.gridLayout.addWidget(self.label_12, 3, 2, 1, 1)

        self.retranslateUi(Monitor)
        QtCore.QObject.connect(self.horizontalSlider, QtCore.SIGNAL(_fromUtf8("valueChanged(int)")), self.lineEdit.copy)
        QtCore.QMetaObject.connectSlotsByName(Monitor)
        Monitor.setTabOrder(self.horizontalSlider, self.lineEdit)

    def retranslateUi(self, Monitor):
        Monitor.setWindowTitle(_translate("Monitor", "RTA Monitor", None))
        self.label_5.setText(_translate("Monitor", "Sim delay", None))
        self.label_14.setText(_translate("Monitor", "0", None))
        self.label_10.setText(_translate("Monitor", "MB/s", None))
        self.label_9.setText(_translate("Monitor", "0", None))
        self.label_8.setText(_translate("Monitor", "RTA rate", None))
        self.label_4.setText(_translate("Monitor", "EBSim rate", None))
        self.label_6.setText(_translate("Monitor", "0", None))
        self.label_7.setText(_translate("Monitor", "MB/s", None))
        self.label_11.setText(_translate("Monitor", "camera/s", None))
        self.label_15.setText(_translate("Monitor", "0", None))
        self.label_12.setText(_translate("Monitor", "event/s", None))

